File: gomutil.cxx
#include "ptoc.h"

/*Sabine Wolf,5999219,q5999219@bonsai.fernuni-hagen.de*/

/*In dieser Unit stecken nuetzliche kleine Funktionen, die an vielen
 Stellen gebraucht werden.*/

#define __gomutil_implementation__


#include "GOMUTIL.h"


integer sgn(integer x)
{
integer sgn_result;
if (x<0) 
sgn_result=-1;
else if (x>0) 
sgn_result=1;
else
sgn_result=0;
return sgn_result;
}

string intstr(integer i )
{
    string s;

   string intstr_result;
   str(i,s);
   intstr_result=s;
   return intstr_result;
}    /* intstr */

integer min(integer a,integer b)
{
   integer min_result;
   if (a>b)  min_result=b; else min_result=a;
   return min_result;
}    /* min */





real minr(real a,real b)
{
   real minr_result;
   if (a>b)  minr_result=b; else minr_result=a;
   return minr_result;
}    /* minr */





integer max(integer a,integer b)
{
   integer max_result;
   if (a<b)  max_result=b; else max_result=a;
   return max_result;
}    /* max */

real maxr(real a,real b)
{
   real maxr_result;
   if (a<b)  maxr_result=b; else maxr_result=a;
   return maxr_result;
}    /* maxr */




class unit_gomutil_initialize {
  public: unit_gomutil_initialize();
};
static unit_gomutil_initialize gomutil_constructor;

unit_gomutil_initialize::unit_gomutil_initialize() {;
}

File: gomutil.pas
{Sabine Wolf,5999219,q5999219@bonsai.fernuni-hagen.de}

{In dieser Unit stecken nuetzliche kleine Funktionen, die an vielen
 Stellen gebraucht werden.}

unit gomutil;

interface

{gibt das Vorzeichen einer Zahl als -1,0 oder 1 zurueck}

function sgn(x:integer):integer;

{Gibt ein integer als string zurueck.}

function intstr(i: integer ):   string;


{Gibt den kleineren wert zurueck.}

function min(a,b : integer):integer;


{Gibt den kleineren Real-Wert zurueck.}

function minr(a,b : real):real;


{Gibt den groesseren wert zurueck.}

function max(a,b : integer):integer;
function maxr(a,b : real):real;

implementation

function sgn(x:integer):integer;
begin
if x<0 then
sgn:=-1
else if x>0 then
sgn:=1
else
sgn:=0;
end;

function intstr(i: integer ):   string;
var s : string;
begin
   str(i,s);
   intstr:=s;
end; { intstr }

function min(a,b : integer):integer;
begin
   if a>b then min:=b else min:=a;
end; { min }





function minr(a,b : real):real;
begin
   if a>b then minr:=b else minr:=a;
end; { minr }





function max(a,b : integer):integer;
begin
   if a<b then max:=b else max:=a;
end; { max }

function maxr(a,b : real):real;
begin
   if a<b then maxr:=b else maxr:=a;
end; { maxr }




begin
end.
